### PR TITLE
Dockerfile rework

### PR DESCRIPTION
**Pull request by mutantcornholio**
* Negative approach to .dockerignore
* Reintruduction of turbo
* Removed unnecessary apt-get calls

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,22 +1,13 @@
...
-.gitignore
-.idea
-combined.log
-docker-compose.yml
-node_modules/
-
-*/**/node_modules/
\ No newline at end of file
+*
+!apps
```
Comment (ironoa) [inline .dockerignore:2]: apps is argocd related, shouldn't be part of the docker image

File: .dockerignore [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
+!candidates
```
Comment (ironoa) [inline .dockerignore:3]: candidates are not necessary in the docker image, they are http fetched or passed via config file

File: .dockerignore [hunk 3]
```diff
@@ -1,22 +1,13 @@
...
+!docs
+!packages
+!scripts
+!package.json
+!tsconfig.json
+!turbo.json
+!.yarnrc.yml
+!.yarn
+!yarn.lock
+.yarn/cache
```
Comment (ironoa) [inline .dockerignore:13]: .yarn/cache ?

Comment (mutantcornholio) [reply .dockerignore:13]: We need `.yarn/releases` and `.yarn/plugins`, but we don't need `.yarn/cache`. 
However, now I think that it's more appropriate to add `!.yarn/releases` and `!.yarn/plugins` directly, and omit ignoring `.yarn/cache`

**Comment by mutantcornholio** [comment]
closed in favor of https://github.com/w3f/1k-validators-be/pull/2939, will address all of the comments there

**Pull request closed without merging**